pymc3/distributions/discrete.py

```diff
@@ -285,8 +285,13 @@ def __init__(self, p, *args, **kwargs):
     def logp(self, value):
         p = self.p
         k = self.k
-
-        return bound(log(p[value]),
+        
+        try:
+            pv = log(p[value])
+        except IndexError:
+            pv = log(constant(p)[value])
```

Comment:
Gotcha. I suggest using `as_tensor_variable` in the constructor. That
should handle all cases.

On Sat, May 30, 2015 at 3:42 PM, Chris Fonnesbeck <notifications@github.com>
wrote:

> In pymc3/distributions/discrete.py
> <https://github.com/pymc-devs/pymc3/pull/718#discussion_r31385239>:
>
> > @@ -285,8 +285,13 @@ def __init__(self, p, *args, **kwargs):
> >      def logp(self, value):
> >          p = self.p
> >          k = self.k
> > -
> > -        return bound(log(p[value]),
> > +
> > +        try:
> > +            pv = log(p[value])
> > +        except IndexError:
> > +            pv = log(constant(p)[value])
>
> If the value is a variable, it cannot be used to index a Numpy array, so
> it generates an index error. This is only for the case where p is a numpy
> array or list and value is a Tensor.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pymc-devs/pymc3/pull/718/files#r31385239>.
>